oak-run/src/main/java/org/apache/jackrabbit/oak/run/DataStoreCommand.java

```diff
@@ -428,15 +433,20 @@ private static void logCliArgs(String[] args) {
         private final NodeStore nodeStore;
         private final String[] paths;
         private final String[] inclusionRegex;
+        private boolean useDirListing;
 
         public NodeTraverserReferenceRetriever(NodeStore nodeStore) {
-            this(nodeStore, null, null);
+            this(nodeStore, null, null, false);
```

Comment:
Why not simply enable useDirListing by default?

Reply:
Because it may change behaviour. There may be arbitrary content with such a property, which then impacts how the tree is traversed. With the new option you explicitly allow the command to take this short cut and kind of give your consent that you are aware of the consequences.